Descr: Adding the Gold team's notebook: MAST Bulk Download through AWS.

yli12313 opened the pull request:
This is the Gold team's bulk download notebook that will undergo the formal notebook verification and publication process!

ttdu commented:
To-dos:

- [ ] Address PEP8 checks
- [ ] Rename to bulk_downloads.ipynb
- [ ] This is conceptually very similar to the [large downloads notebook](https://github.com/spacetelescope/mast_notebooks/tree/main/notebooks/multi_mission/large_downloads). It may be reasonable to combine them. Who requested this notebook / what are the requirements?
- [ ] requirements should specify a version of astroquery, rather than astropy (I think this is causing the validation check to fail)
- [ ] link updates: change the linked text "here" ([this is bad](https://uctechnews.ucop.edu/no-more-click-here-why-descriptive-link-text-is-important/) for people who use screen readers)). maybe add a few more links, e.g. to MAST
- [ ] we should define what we mean by "bulk", and why people should prefer this method. are we just trying to lighten the load on MAST servers? do we anticipate a use-case where people pull all of the GALEX/PS1 data? for that matter, why do we call out those specific missions? HST almost certainly has more data (by GB) than GALEX, so why not focus on that?

yli12313 commented:
@ttdu: Thank you for the feedback. It's good to know that there's another notebook that's similar and it's better to combine both notebooks. Some of the questions you posed are things we would need to address on the Gold team. We have never gone through the formal notebook validation and publication process before, and thought our notebook would be of value to the scientists. You make a lot of good points that we did not address, so I think we will start addressing each point either next sprint or the sprint after. Time is hard to come by right now given that B. Jordan on our team is on leave. But once things settle down, I will address everything such that we have a notebook that's actually useful. Thank you! - Y

ttdu commented:
No worries -- we can take as long as needed for the review. Your notebook definitely has value to the scientists! I hope my comments did not convince you otherwise.

Happy to help if you have questions about the particulars.

yli12313 commented:
@ttdu: The Gold team is picking up this work again starting this week or next week. I will start addressing some of these errors that were picked up by the testing. Thank you!

bshiao commented:
@ttdu WRT to 
"This is conceptually very similar to the [large downloads notebook](https://github.com/spacetelescope/mast_notebooks/tree/main/notebooks/multi_mission/large_downloads). It may be reasonable to combine them. Who requested this notebook / what are the requirements?"

We've gotten requests from the users (help desk tickets) to download large chunks of GALEX data so it seemed to make sense to create a NB which serves that community in particular. The same applies to PanSTARRS.
Since these data are in AWS it makes the most sense for users to get it from there using these NB as opposed to our normal UI which in many cases don't offer the capability to large downloads of particular file types across the entire sky (as one example).

bshiao commented:
@ttdu Yingquan and I will define this better in the NB or should it be in the description, not sure where is the best place to define this? 

We are trying to lighten the load on the MAST servers and also provide users with a hopefully simple way of obtaining large swaths of both GALEX and PanSTARRS data either based on some large region of the sky (cone or box)  or in the case of GALEX users they want a certain file type for the entire mission which is small enough for the community to retrieve. While it's possible to do that for GALEX now it's a special case request that usually I handle by providing them with a curl/wget script and if we can do away with me being in the middle by providing a NB that's probably a good thing.

If we can eventually create a NB that is agnostic of mission, i.e. say give me the main image files for GALEX in this region of the sky or PanSTARRS for these bands or whatever that's great but in the meantime individual NB seem a good alternative?

"we should define what we mean by "bulk", and why people should prefer this method. are we just trying to lighten the load on MAST servers? do we anticipate a use-case where people pull all of the GALEX/PS1 data? for that matter, why do we call out those specific missions? HST almost certainly has more data (by GB) than GALEX, so why not focus on that?"

yli12313 commented:
**DONE**:
- [x]:  Given that the directory name and notebook name should match, the rename is *Bulk_Download_Gold.ipynb*.

yli12313 commented:
@ttdu: I have some questions that I will bring up on Monday.

- How do we address these PEP8 errors.
- I made some NB updates and the PR updated. How do I get the CI/CD checks to re-run?

Thank you!

yli12313 commented:
Error 1 - I can fix this:
`Run python temp_pep8checker/.github/helpers/pep8_nb_style_checker.py notebooks/multi_mission/Bulk_Download_Gold/Bulk_Download_Gold.ipynb
PEP8 error 1 of 3
PEP8 error found in code cell 3 (Notebook cell [12](https://github.com/spacetelescope/mast_notebooks/actions/runs/13860959551/job/38793550524#step:7:13)) at code cell line 5, column 45
obs = Observations.query_criteria(s_ra=[30.2,31.2], s_dec=[-10.25,-9.25], obs_collection="GALEX")
                                            ▲
E231 missing whitespace after ','`

Error 2 - I think I can simply import boto3 and it should be good, no? Let me push another commit to PR.
`ModuleNotFoundError: No module named 'boto3'`

yli12313 commented:
@ttdu: I fixed the white space error and added boto3. Would if be possible to run the GitHub Actions workflow again? Think I can't trigger it as a non-maintainer. Thank you!

adrianlucy commented:
re. the failing checks, I think you may just need to add boto3 to the requirements.txt file. I suspect the execution errors all stem from boto3 not being installed. Try that?

Incidentally, you don't need to explicitly `import boto3`. You can delete that line; it will be imported implicitly by astroquery.

Our team needs to do some copyediting on this before we publish it; I emailed you to make sure our timelines are compatible.

yli12313 commented:
@adrianlucy: I addressed all the errors that you picked up; thank you, the comments were very helpful. I hope there's no errors now (fingers crossed!).

adrianlucy commented:
Yep, all CI checks passed! @yli12313, as discussed we'll work on copyediting the week of 21st or earlier. We'll probably make those changes ourselves — so in order to avoid having to resolve conflicts between commits, please give us a heads-up if you plan to make further refinements.

adrianlucy commented:
Oh, @yli12313 please make a new commit with all code cell outputs cleared ("Clear Outputs"), which is [the preferred format for this repository](https://github.com/spacetelescope/mast_notebooks/blob/main/contributing/CONTRIBUTING.md). We could do that on our end but I fear we might forget to make it a separate commit, which would make git diffs a hassle.

(For future reference, if a notebook had output that was taking up a lot of space I'd ask you to amend the old commit due to space considerations. Fortunately, this notebook is tiny and only has text output, so the commit history is not an issue here.)

yli12313 commented:
@adrianlucy: I did it. Thank you for all your help!

adrianlucy commented:
Thank you!

adrianlucy commented:
Hi @ttdu, please review the latest revision, and then afterwards let Yingquan know it is ready for his final review. Thanks!

adrianlucy commented:
(Just made a further minor commit @ttdu, apologies if you were already revising it.)

yli12313 commented:
@ttdu, @adrianlucy: Thank you for updating the notebook and for merging the this PR into main!! Thank you for all your help!

Quick question: When can we see the notebook on the [MAST Notebook Repository](https://spacetelescope.github.io/mast_notebooks/intro.html)? Thank you!
